src/Receiver/NotificationReceiver.php

```diff
@@ -0,0 +1,178 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Webhook Module for PHP
+ *
+ * Copyright (c) 2021 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Webhook\Receiver;
+
+use Adyen\AdyenException;
+use Adyen\Exception\AuthenticationException;
+use Adyen\Webhook\Exception\HMACKeyValidationException;
+use Adyen\Exception\MerchantAccountCodeException;
+use Adyen\Webhook\Exception\InvalidDataException;
+
+class NotificationReceiver
+{
+    /**
+     * @var HmacSignature
+     */
+    private $hmacSignature;
+
+    /**
+     * NotificationReceiver constructor.
+     * @param HmacSignature $hmacSignature
+     */
+    public function __construct(
+        HmacSignature $hmacSignature
+    ) {
+        $this->hmacSignature = $hmacSignature;
+    }
+
+    /**
+     * Checks if the hmac key is valid
+     * @param $response
+     * @param $hmacKey
+     * @return bool
+     * @throws InvalidDataException
+     * @throws HMACKeyValidationException
+     */
+    public function validateHmac($response, $hmacKey)
+    {
+        $isTestNotification = $this->isTestNotification($response['pspReference']);
+        if (!$this->hmacSignature->isValidNotificationHMAC($hmacKey, $response)) {
+            if ($isTestNotification) {
+                $message = 'HMAC key validation failed';
+                throw new HMACKeyValidationException($message);
+            }
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * @param $response
+     * @param $merchantAccount
+     * @param $notificationUsername
+     * @param $notificationPassword
+     * @return bool
+     * @throws AuthenticationException
+     * @throws MerchantAccountCodeException
+     */
+    public function isAuthenticated($response, $merchantAccount, $notificationUsername, $notificationPassword)
```

Comment:
Would it be a good idea to actually provide a request object instead of just using the `$_SERVER['PHP_AUTH_USER']` and  `$_SERVER['PHP_AUTH_PW']`? In case of different server types these server variables can differ. Not needed to fix in this PR but I think a follow up ticket could worth a shot to optimise this piece a bit more. What do you think?

Reply:
I think this indeed could be an improvement, although we need to investigate the impact because maybe not everyone uses a [PSR-7](https://www.php-fig.org/psr/psr-7/) compliant request object like `Symfony\Component\HttpFoundation\Request` which shopware6 uses. For this PR I want to keep is as simple as possible so that I can just update the sw6 plugin (and others) to switch to this module without changing much else.
But I can make a ticket for investigating the impact of adding a http lib as a dependency of this module

Reply:
let's do it and merge this PR asap without this indeed!